src/Microsoft.Identity.Client/Platforms/iOS/iOSBroker.cs

```diff
@@ -0,0 +1,236 @@
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using Microsoft.Identity.Client.Core;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Utils;
+using UIKit;
+using Foundation;
+using System;
+using CoreFoundation;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Exceptions;
+using Microsoft.Identity.Client.ApiConfig;
+
+namespace Microsoft.Identity.Client.Platforms.iOS
+{
+    /// <summary>
+    /// Handles requests which invoke the broker. This is only for mobile (iOS and Android) scenarios.
+    /// </summary>
+    internal class iOSBroker : NSObject, IBroker
+    {
+        private static SemaphoreSlim brokerResponseReady = null;
+        private static NSUrl brokerResponse = null;
+
+        private Dictionary<string, string> _brokerPayload;
+
+        private readonly ICoreLogger _logger;
+        private IServiceBundle _serviceBundle;
+
+        public iOSBroker(ICoreLogger logger)
+        {
+            _logger = logger;
+        }
+
+        public bool CanInvokeBroker(OwnerUiParent uiParent, IServiceBundle serviceBundle)
+        {
+            _serviceBundle = serviceBundle;
+
+            if (uiParent == null)
+            {
+                _logger.Verbose(iOSBrokerConstants.UiParentIsNullCannotInvokeBroker);
+                return false;
+            }
+
+            if (uiParent.CoreUiParent.CallerViewController == null)
+            {
+                _logger.Verbose(iOSBrokerConstants.CallerViewControllerIsNullCannotInvokeBroker);
+                return false;
+            }
+
+            var result = false;
+
+            if (_serviceBundle.Config.IsBrokerEnabled)
```

Comment:
>if (_serviceBundle.Config.IsBrokerEnabled) [](start = 12, length = 42)

consider inverting and returning if not enabled.

Reply:
Need to have the else statement.


---
In reply to: [256626333](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r256626333) [](ancestors = 256626333)